Queueomatic/Server/Endpoints/Hubs/Room/RoomHub.cs

```diff
@@ -1,24 +1,42 @@
 ﻿using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Caching.Memory;
+using Queueomatic.DataAccess.Models;
+using Queueomatic.Server.Services.CacheRoomService;
 using Queueomatic.Shared.DTOs;
+using Queueomatic.Shared.Models;
 
 namespace Queueomatic.Server.Endpoints.Hubs.Room;
 
 public class RoomHub : Hub
 {
+    private readonly ICacheRoomService _cacheService;
+    public RoomHub(ICacheRoomService cacheService)
+    {
+        _cacheService = cacheService ?? throw new ArgumentException($"The value of cache cannot be null");
+    }
+
     public async Task UpdateParticipant(ParticipantRoomDto participant, StatusDto status, string roomId)
     {
-        participant.StatusDate = DateTime.UtcNow;
-        await Clients.Groups(roomId).SendAsync("MoveParticipant", participant, status);
+        var participantToBeUpdated = _cacheService.UpdateRoom(participant, status, roomId);
+        await Clients.Groups(roomId).SendAsync("MoveParticipant", participantToBeUpdated, status);
+    }
+
+    public async Task JoinRoom(string roomName)
+    {
+        await Groups.AddToGroupAsync(Context.ConnectionId, roomName);
     }
 
-    public Task JoinRoom(string roomName)
+    public async Task<RoomModel> InitializeParticipant(ParticipantRoomDto participant, string roomId, string roomName)
     {
-        return Groups.AddToGroupAsync(Context.ConnectionId, roomName);
+        if (_cacheService.InitRoom(roomId, roomName, participant))
+            await Clients.Groups(roomId).SendAsync("MoveParticipant", participant, Status.Idling);
```

Comment:
I assume this line is responsible for adding a new participant to either of the lists of the room. But this will be hit only once, only the first time.

Reply:
Yes, and this was incorrect. Changed in new commit.